lib/chess/core/book.rb

```diff
@@ -10,121 +10,21 @@ class Book
 
       def initialize(board)
         @board = board
-        @cells = board.cells
-        @pieces = board.pieces
-        @record = Record.new
+        @record = Chess::Core::Record.new
+        @move = Chess::Core::Move::Main
```

Comment:
Because each time I'll be using an instance of `Chess::Core::Book` with `#move` I'm instancing a move, so I was thinking is more clear if I save `Chess::Core::Move::Main` in a variable and use it when Book's instance instantiates `Chess::Core::Move::Main`:

```ruby
move = @move.new(
  intention,
  @board,
  team_playing
)
```

Reply:
ok cool, the problem is that `Chess::Core::Move::Main` isn't quite an attribute of the `Book` class. I think it's better to just  call the class directly instead of having this attribute. Also, you are just applying this pattern here and you're not being consistent on other places